### PR TITLE
Add sems-rhel7 env with clang, gnu, and intel compilers (TRIL-236)

### PR DESCRIPTION
**Pull request by bartlettroscoe**
The sems-rhel7 env should be selected automatically on any SNL CEO RHEL7
machine that has the SEMS modules mounted.

Currently, the intel-17.0.1 compilers are super slow so this build is
basically unusable (at least on cortado).

There is also the start of a cuda 9.2 + GCC 7.2.0 build I think we need to
know the KOKKOS_ARCH for the GPUs on this machine for this to work.

As part of this, I improved the error messages some for what complilers are
supported.

### CODE REVIEW

**Comment by bartlettroscoe** [comment]
CC: @fryeguy52, @jmgate 

@bathmatt, I am running some local builds on 'cortado' but it looks like the `clang-3.9.0` and `gnu-7.2.0` builds are working okay.  The `intel-17.0.1` build is unusable because the compiler is super slow (must be problems communicating with Intel license sever or something).  Also, I think I need to know that `KOKKOS_ARCH` for this system.

Can you please give this a try by pulling the branch and trying this out?

**Comment by bartlettroscoe** [comment]
On 'cortado', I ran:

```
$ ./checkin-test-atdm.sh \
   clang-3.9.0-openmp-release-debug \
   gnu-7.2.0-openmp-release-debug \
   intel-17.0.1-openmp-release-debug \
   --enable-packages=Kokkos,Teuchos,Panzer --local-do-all
```

and it returned:

```
PASSED (NOT READY TO PUSH): Trilinos: cortado.sandia.gov

Sat Dec  8 00:38:29 MST 2018

Enabled Packages: Kokkos, Teuchos, Panzer

Build test results:
-------------------
1) clang-3.9.0-openmp-release-debug => passed: passed=334,notpassed=0 (17.46 min)
2) gnu-7.2.0-openmp-release-debug => passed: passed=334,notpassed=0 (22.41 min)
3) intel-17.0.1-openmp-release-debug => passed: passed=334,notpassed=0 (220.26 min)
```

So the intel-17.0.1 build does work, it is just incredibly slow.  Here are the timings for the intel-17.0.1 build:

```
1) intel-17.0.1-openmp-release-debug Results:
---------------------------------------------

  passed: Trilinos/intel-17.0.1-openmp-release-debug: passed=334,notpassed=0
  
  Sat Dec  8 00:38:27 MST 2018
  
  Enabled Packages: Kokkos, Teuchos, Panzer
  Hostname: cortado.sandia.gov
  Source Dir: /home/rabartl/Trilinos.base/Trilinos/cmake/tribits/ci_support/../../..
  Build Dir: /home/rabartl/Trilinos.base/BUILDS/ATDM/SEMS_RHEL7/CHECKIN/intel-17.0.1-openmp-release-debug
  
  CMake Cache Varibles: -GNinja -DTrilinos_TRIBITS_DIR:PATH=/home/rabartl/Trilinos.base/Trilinos/cmake/tribits -DTrilinos_ENABLE_TESTS:BOOL=ON -DTrilinos_TEST_CATEGORIES:STRING=NIGHTLY -DTrilinos_ALLOW_NO_PACKAGES:BOOL=OFF -DDART_TESTING_TIMEOUT:STRING=600.0 -GNinja -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake -DTrilinos_TRACE_ADD_TEST=ON -DTrilinos_ENABLE_Kokkos:BOOL=ON -DTrilinos_ENABLE_Teuchos:BOOL=ON -DTrilinos_ENABLE_Panzer:BOOL=ON -DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=ON -DTrilinos_ENABLE_ALL_FORWARD_DEP_PACKAGES:BOOL=OFF
  Make Options: -j 20
  CTest Options: -j 10
  
  Pull: Not Performed
  Configure: Passed (49.17 min)
  Build: Passed (168.66 min)
  Test: Passed (2.43 min)
  
  100% tests passed, 0 tests failed out of 334
  
  Subproject Time Summary:
  Kokkos     =  89.74 sec*proc (27 tests)
  Panzer     = 977.67 sec*proc (170 tests)
  Teuchos    = 105.14 sec*proc (137 tests)
  
  Total Test time (real) = 145.96 sec
  
  Total time for intel-17.0.1-openmp-release-debug = 220.26 min
```

That is 49 minutes to configure!!!!!

I am working on the cuda-9.2 build now.

**Comment by bartlettroscoe** [comment]
CC: @bathmatt, @jmgate, @fryeguy52 

I have this env pretty well set up.  The Intel build is unusuable due to a problem communicating with the Intel license server (see [TRIL-236](https://sems-atlassian-son.sandia.gov/jira/browse/TRIL-236?focusedCommentId=24969&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-24969)) but the clang-3.9.0, gnu-7.2.0, and cuda-9.2 (with gnu-7.2.0) builds seem to be working just fine.  We seem some test failures on the ATDM Trilinos test suite but I think we have seem most of these failures and they are pretty minor.

----

Running a bunch of builds with:

```
$ ./checkin-test-atdm.sh all --enable-all-packages=on --local-do-all

***
*** ./checkin-test-atdm.sh  all --enable-all-packages=on --local-do-all
***

ATDM_TRILINOS_DIR = '/home/rabartl/Trilinos.base/Trilinos/cmake/std/atdm/../../..'

Load some env to get python, cmake, etc ...

Hostname 'cortado.sandia.gov' matches known ATDM host 'sems-rhel7' and system 'sems-rhel7'
Setting compiler and build options for buld name 'default'
Using SEMS RHEL7 compiler stack GNU-7.2.0 to build DEBUG code with Kokkos node type SERIAL
File local-checkin-test-defaults.py already exists, leaving it!

Running configure, build, and/or testing for 6 builds: sems-rhel7-clang-3.9.0-openmp-release-debug sems-rhel7-clang-3.9.0-serial-release-debug sems-rhel7-gnu-7.2.0-openmp-release-debug sems-rhel7-gnu-7.2.0-serial-release-debug sems-rhel7-cuda-9.2-Pascal60-opt sems-rhel7-cuda-9.2-Pascal60-release-debug
```

This gave the result:

```
FAILED (NOT READY TO PUSH): Trilinos: cortado.sandia.gov

Sun Dec  9 19:38:00 MST 2018

Enabled Packages: 

Build test results:
-------------------
1) sems-rhel7-clang-3.9.0-openmp-release-debug => FAILED: passed=1993,notpassed=2 => Not ready to push! (13.06 min)
2) sems-rhel7-clang-3.9.0-serial-release-debug => FAILED: passed=1884,notpassed=2 => Not ready to push! (12.14 min)
3) sems-rhel7-gnu-7.2.0-openmp-release-debug => passed: passed=1995,notpassed=0 (9.05 min)
4) sems-rhel7-gnu-7.2.0-serial-release-debug => FAILED: passed=1885,notpassed=1 => Not ready to push! (57.69 min)
5) sems-rhel7-cuda-9.2-Pascal60-opt => FAILED: passed=1999,notpassed=1 => Not ready to push! (159.15 min)
6) sems-rhel7-cuda-9.2-Pascal60-release-debug => FAILED: build failed => Not ready to push! (74.48 min)


REQUESTED ACTIONS: FAILED
```

<details>
<summary><b>DETAILED BUILD AND TEST RESULTS</b> (click to expand)</summary>

```
1) sems-rhel7-clang-3.9.0-openmp-release-debug Results:
-------------------------------------------------------

  FAILED: Trilinos/sems-rhel7-clang-3.9.0-openmp-release-debug: passed=1993,notpassed=2
  
  Sun Dec  9 14:25:01 MST 2018
  
  Enabled Packages: 
  Enabled all Packages
  Hostname: cortado.sandia.gov
  Source Dir: /home/rabartl/Trilinos.base/Trilinos/cmake/tribits/ci_support/../../..
  Build Dir: /home/rabartl/Trilinos.base/BUILDS/ATDM/SEMS_RHEL7/CHECKIN/sems-rhel7-clang-3.9.0-openmp-release-debug
  
  CMake Cache Varibles: -GNinja -DTrilinos_TRIBITS_DIR:PATH=/home/rabartl/Trilinos.base/Trilinos/cmake/tribits -DTrilinos_ENABLE_TESTS:BOOL=ON -DTrilinos_TEST_CATEGORIES:STRING=NIGHTLY -DTrilinos_ALLOW_NO_PACKAGES:BOOL=OFF -DDART_TESTING_TIMEOUT:STRING=600.0 -GNinja -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake -DTrilinos_TRACE_ADD_TEST=ON -DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=ON -DTrilinos_ENABLE_ALL_PACKAGES:BOOL=ON -DTrilinos_ENABLE_ALL_FORWARD_DEP_PACKAGES:BOOL=OFF
  Make Options: -j 20
  CTest Options: -j 10
  
  Pull: Not Performed
  Configure: Passed (0.94 min)
  Build: Passed (3.95 min)
  Test: FAILED (8.17 min)
  
  99% tests passed, 2 tests failed out of 1995
  
  Subproject Time Summary:
  Amesos2          =   8.59 sec*proc (7 tests)
  Anasazi          = 115.38 sec*proc (74 tests)
  Belos            = 157.68 sec*proc (96 tests)
  Ifpack2          =  55.04 sec*proc (45 tests)
  Intrepid2        = 229.83 sec*proc (260 tests)
  Kokkos           =  93.27 sec*proc (27 tests)
  KokkosKernels    = 212.14 sec*proc (8 tests)
  MueLu            = 533.07 sec*proc (99 tests)
  NOX              = 118.42 sec*proc (105 tests)
  Panzer           = 1829.99 sec*proc (170 tests)
  Phalanx          =  11.78 sec*proc (27 tests)
  Piro             =  15.18 sec*proc (12 tests)
  Rythmos          =  41.09 sec*proc (83 tests)
  SEACAS           =  11.91 sec*proc (20 tests)
  STK              =   1.68 sec*proc (3 tests)
  Sacado           = 104.02 sec*proc (297 tests)
  Stratimikos      =  24.20 sec*proc (39 tests)
  Teko             =  54.38 sec*proc (19 tests)
  Tempus           = 141.59 sec*proc (80 tests)
  Teuchos          =  94.66 sec*proc (137 tests)
  Thyra            =  62.11 sec*proc (81 tests)
  Tpetra           = 234.98 sec*proc (187 tests)
  Xpetra           =  29.16 sec*proc (18 tests)
  Zoltan2          = 172.13 sec*proc (101 tests)
  
  Total Test time (real) = 489.79 sec
  
  The following tests FAILED:
  	1154 - Teko_ModALPreconditioner_MPI_1 (Failed)
  	1572 - MueLu_UnitTestsBlockedEpetra_MPI_1 (Failed)
  Errors while running CTest
  
  Total time for sems-rhel7-clang-3.9.0-openmp-release-debug = 13.06 min


2) sems-rhel7-clang-3.9.0-serial-release-debug Results:
-------------------------------------------------------

  FAILED: Trilinos/sems-rhel7-clang-3.9.0-serial-release-debug: passed=1884,notpassed=2
  
  Sun Dec  9 14:37:15 MST 2018
  
  Enabled Packages: 
  Enabled all Packages
  Hostname: cortado.sandia.gov
  Source Dir: /home/rabartl/Trilinos.base/Trilinos/cmake/tribits/ci_support/../../..
  Build Dir: /home/rabartl/Trilinos.base/BUILDS/ATDM/SEMS_RHEL7/CHECKIN/sems-rhel7-clang-3.9.0-serial-release-debug
  
  CMake Cache Varibles: -GNinja -DTrilinos_TRIBITS_DIR:PATH=/home/rabartl/Trilinos.base/Trilinos/cmake/tribits -DTrilinos_ENABLE_TESTS:BOOL=ON -DTrilinos_TEST_CATEGORIES:STRING=NIGHTLY -DTrilinos_ALLOW_NO_PACKAGES:BOOL=OFF -DDART_TESTING_TIMEOUT:STRING=600.0 -GNinja -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake -DTrilinos_TRACE_ADD_TEST=ON -DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=ON -DTrilinos_ENABLE_ALL_PACKAGES:BOOL=ON -DTrilinos_ENABLE_ALL_FORWARD_DEP_PACKAGES:BOOL=OFF
  Make Options: -j 20
  CTest Options: -j 10
  
  Pull: Not Performed
  Configure: Passed (0.87 min)
  Build: Passed (3.54 min)
  Test: FAILED (7.73 min)
  
  99% tests passed, 2 tests failed out of 1886
  
  Subproject Time Summary:
  Amesos2          =   7.84 sec*proc (7 tests)
  Anasazi          = 101.67 sec*proc (74 tests)
  Belos            = 145.62 sec*proc (96 tests)
  Ifpack2          =  51.02 sec*proc (45 tests)
  Intrepid2        = 152.34 sec*proc (160 tests)
  Kokkos           =  42.10 sec*proc (24 tests)
  KokkosKernels    = 111.13 sec*proc (4 tests)
  MueLu            = 532.22 sec*proc (96 tests)
  NOX              = 100.41 sec*proc (105 tests)
  Panzer           = 1847.09 sec*proc (175 tests)
  Phalanx          =  11.34 sec*proc (27 tests)
  Piro             =  14.72 sec*proc (12 tests)
  Rythmos          =  40.04 sec*proc (83 tests)
  SEACAS           =  11.81 sec*proc (20 tests)
  STK              =   1.75 sec*proc (3 tests)
  Sacado           =  96.29 sec*proc (293 tests)
  Stratimikos      =  22.09 sec*proc (39 tests)
  Teko             =  69.09 sec*proc (19 tests)
  Tempus           = 136.63 sec*proc (80 tests)
  Teuchos          =  94.84 sec*proc (137 tests)
  Thyra            =  59.90 sec*proc (81 tests)
  Tpetra           = 226.73 sec*proc (187 tests)
  Xpetra           =  26.19 sec*proc (18 tests)
  Zoltan2          = 163.95 sec*proc (101 tests)
  
  Total Test time (real) = 463.64 sec
  
  The following tests FAILED:
  	1143 - Teko_ModALPreconditioner_MPI_1 (Failed)
  	1461 - MueLu_UnitTestsBlockedEpetra_MPI_1 (Failed)
  Errors while running CTest
  
  Total time for sems-rhel7-clang-3.9.0-serial-release-debug = 12.14 min


3) sems-rhel7-gnu-7.2.0-openmp-release-debug Results:
-----------------------------------------------------

  passed: Trilinos/sems-rhel7-gnu-7.2.0-openmp-release-debug: passed=1995,notpassed=0
  
  Sun Dec  9 14:46:23 MST 2018
  
  Enabled Packages: 
  Enabled all Packages
  Hostname: cortado.sandia.gov
  Source Dir: /home/rabartl/Trilinos.base/Trilinos/cmake/tribits/ci_support/../../..
  Build Dir: /home/rabartl/Trilinos.base/BUILDS/ATDM/SEMS_RHEL7/CHECKIN/sems-rhel7-gnu-7.2.0-openmp-release-debug
  
  CMake Cache Varibles: -GNinja -DTrilinos_TRIBITS_DIR:PATH=/home/rabartl/Trilinos.base/Trilinos/cmake/tribits -DTrilinos_ENABLE_TESTS:BOOL=ON -DTrilinos_TEST_CATEGORIES:STRING=NIGHTLY -DTrilinos_ALLOW_NO_PACKAGES:BOOL=OFF -DDART_TESTING_TIMEOUT:STRING=600.0 -GNinja -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake -DTrilinos_TRACE_ADD_TEST=ON -DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=ON -DTrilinos_ENABLE_ALL_PACKAGES:BOOL=ON -DTrilinos_ENABLE_ALL_FORWARD_DEP_PACKAGES:BOOL=OFF
  Make Options: -j 20
  CTest Options: -j 10
  
  Pull: Not Performed
  Configure: Passed (0.90 min)
  Build: Passed (0.03 min)
  Test: Passed (8.12 min)
  
  100% tests passed, 0 tests failed out of 1995
  
  Subproject Time Summary:
  Amesos2          =   9.30 sec*proc (7 tests)
  Anasazi          = 118.17 sec*proc (74 tests)
  Belos            = 155.99 sec*proc (96 tests)
  Ifpack2          =  53.12 sec*proc (45 tests)
  Intrepid2        = 246.59 sec*proc (260 tests)
  Kokkos           =  89.58 sec*proc (27 tests)
  KokkosKernels    = 174.15 sec*proc (8 tests)
  MueLu            = 457.38 sec*proc (99 tests)
  NOX              = 113.78 sec*proc (105 tests)
  Panzer           = 1687.06 sec*proc (170 tests)
  Phalanx          =  21.38 sec*proc (27 tests)
  Piro             =  13.98 sec*proc (12 tests)
  Rythmos          =  38.88 sec*proc (83 tests)
  SEACAS           =  13.04 sec*proc (20 tests)
  STK              =   6.73 sec*proc (3 tests)
  Sacado           =  97.18 sec*proc (297 tests)
  Stratimikos      =  20.97 sec*proc (39 tests)
  Teko             =  37.59 sec*proc (19 tests)
  Tempus           = 138.30 sec*proc (80 tests)
  Teuchos          =  97.81 sec*proc (137 tests)
  Thyra            =  62.45 sec*proc (81 tests)
  Tpetra           = 207.81 sec*proc (187 tests)
  Xpetra           =  32.94 sec*proc (18 tests)
  Zoltan2          = 172.10 sec*proc (101 tests)
  
  Total Test time (real) = 486.99 sec
  
  Total time for sems-rhel7-gnu-7.2.0-openmp-release-debug = 9.05 min


4) sems-rhel7-gnu-7.2.0-serial-release-debug Results:
-----------------------------------------------------

  FAILED: Trilinos/sems-rhel7-gnu-7.2.0-serial-release-debug: passed=1885,notpassed=1
  
  Sun Dec  9 15:44:09 MST 2018
  
  Enabled Packages: 
  Enabled all Packages
  Hostname: cortado.sandia.gov
  Source Dir: /home/rabartl/Trilinos.base/Trilinos/cmake/tribits/ci_support/../../..
  Build Dir: /home/rabartl/Trilinos.base/BUILDS/ATDM/SEMS_RHEL7/CHECKIN/sems-rhel7-gnu-7.2.0-serial-release-debug
  
  CMake Cache Varibles: -GNinja -DTrilinos_TRIBITS_DIR:PATH=/home/rabartl/Trilinos.base/Trilinos/cmake/tribits -DTrilinos_ENABLE_TESTS:BOOL=ON -DTrilinos_TEST_CATEGORIES:STRING=NIGHTLY -DTrilinos_ALLOW_NO_PACKAGES:BOOL=OFF -DDART_TESTING_TIMEOUT:STRING=600.0 -GNinja -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake -DTrilinos_TRACE_ADD_TEST=ON -DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=ON -DTrilinos_ENABLE_ALL_PACKAGES:BOOL=ON -DTrilinos_ENABLE_ALL_FORWARD_DEP_PACKAGES:BOOL=OFF
  Make Options: -j 20
  CTest Options: -j 10
  
  Pull: Not Performed
  Configure: Passed (0.84 min)
  Build: Passed (49.38 min)
  Test: FAILED (7.47 min)
  
  99% tests passed, 1 tests failed out of 1886
  
  Subproject Time Summary:
  Amesos2          =   8.47 sec*proc (7 tests)
  Anasazi          = 104.04 sec*proc (74 tests)
  Belos            = 147.33 sec*proc (96 tests)
  Ifpack2          =  51.80 sec*proc (45 tests)
  Intrepid2        = 169.52 sec*proc (160 tests)
  Kokkos           =  47.38 sec*proc (24 tests)
  KokkosKernels    =  96.99 sec*proc (4 tests)
  MueLu            = 427.58 sec*proc (96 tests)
  NOX              = 106.81 sec*proc (105 tests)
  Panzer           = 1637.71 sec*proc (175 tests)
  Phalanx          =   8.76 sec*proc (27 tests)
  Piro             =  14.02 sec*proc (12 tests)
  Rythmos          =  37.78 sec*proc (83 tests)
  SEACAS           =  12.36 sec*proc (20 tests)
  STK              =   1.47 sec*proc (3 tests)
  Sacado           = 109.66 sec*proc (293 tests)
  Stratimikos      =  22.79 sec*proc (39 tests)
  Teko             =  57.70 sec*proc (19 tests)
  Tempus           = 135.79 sec*proc (80 tests)
  Teuchos          = 102.05 sec*proc (137 tests)
  Thyra            =  59.77 sec*proc (81 tests)
  Tpetra           = 215.43 sec*proc (187 tests)
  Xpetra           =  25.73 sec*proc (18 tests)
  Zoltan2          = 163.90 sec*proc (101 tests)
  
  Total Test time (real) = 448.16 sec
  
  The following tests FAILED:
  	924 - Belos_Tpetra_HybridGMRES_hb_test_1_MPI_4 (Failed)
  Errors while running CTest
  
  Total time for sems-rhel7-gnu-7.2.0-serial-release-debug = 57.69 min


5) sems-rhel7-cuda-9.2-Pascal60-opt Results:
--------------------------------------------

  FAILED: Trilinos/sems-rhel7-cuda-9.2-Pascal60-opt: passed=1999,notpassed=1
  
  Sun Dec  9 18:23:24 MST 2018
  
  Enabled Packages: 
  Enabled all Packages
  Hostname: cortado.sandia.gov
  Source Dir: /home/rabartl/Trilinos.base/Trilinos/cmake/tribits/ci_support/../../..
  Build Dir: /home/rabartl/Trilinos.base/BUILDS/ATDM/SEMS_RHEL7/CHECKIN/sems-rhel7-cuda-9.2-Pascal60-opt
  
  CMake Cache Varibles: -GNinja -DTrilinos_TRIBITS_DIR:PATH=/home/rabartl/Trilinos.base/Trilinos/cmake/tribits -DTrilinos_ENABLE_TESTS:BOOL=ON -DTrilinos_TEST_CATEGORIES:STRING=NIGHTLY -DTrilinos_ALLOW_NO_PACKAGES:BOOL=OFF -DDART_TESTING_TIMEOUT:STRING=600.0 -GNinja -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake -DTrilinos_TRACE_ADD_TEST=ON -DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=ON -DTrilinos_ENABLE_ALL_PACKAGES:BOOL=ON -DTrilinos_ENABLE_ALL_FORWARD_DEP_PACKAGES:BOOL=OFF
  Make Options: -j 20
  CTest Options: -j 10
  
  Pull: Not Performed
  Configure: Passed (1.79 min)
  Build: Passed (88.97 min)
  Test: FAILED (68.39 min)
  
  99% tests passed, 1 tests failed out of 2000
  
  Subproject Time Summary:
  Amesos2          = 109.96 sec*proc (8 tests)
  Anasazi          = 697.56 sec*proc (74 tests)
  Belos            = 700.16 sec*proc (96 tests)
  Ifpack2          = 945.32 sec*proc (45 tests)
  Intrepid2        = 920.58 sec*proc (267 tests)
  Kokkos           = 470.82 sec*proc (27 tests)
  KokkosKernels    = 490.31 sec*proc (8 tests)
  MueLu            = 8041.54 sec*proc (99 tests)
  NOX              = 354.61 sec*proc (105 tests)
  Panzer           = 13534.59 sec*proc (162 tests)
  Phalanx          =  32.34 sec*proc (27 tests)
  Piro             =  15.13 sec*proc (12 tests)
  Rythmos          =  34.53 sec*proc (83 tests)
  SEACAS           =  14.14 sec*proc (20 tests)
  STK              =   2.98 sec*proc (3 tests)
  Sacado           = 191.27 sec*proc (300 tests)
  Stratimikos      =  20.87 sec*proc (39 tests)
  Teko             = 1246.70 sec*proc (19 tests)
  Tempus           = 121.95 sec*proc (80 tests)
  Teuchos          =  99.05 sec*proc (137 tests)
  Thyra            =  82.05 sec*proc (81 tests)
  Tpetra           = 3900.02 sec*proc (189 tests)
  Xpetra           = 613.59 sec*proc (18 tests)
  Zoltan2          = 1433.99 sec*proc (101 tests)
  
  Total Test time (real) = 4103.39 sec
  
  The following tests FAILED:
  	1977 - PanzerAdaptersSTK_MixedPoissonExample-ConvTest-Hex-Order-3 (Timeout)
  Errors while running CTest
  
  Total time for sems-rhel7-cuda-9.2-Pascal60-opt = 159.15 min


6) sems-rhel7-cuda-9.2-Pascal60-release-debug Results:
------------------------------------------------------

  FAILED: Trilinos/sems-rhel7-cuda-9.2-Pascal60-release-debug: build failed
  
  Sun Dec  9 19:37:58 MST 2018
  
  Enabled Packages: 
  Enabled all Packages
  Hostname: cortado.sandia.gov
  Source Dir: /home/rabartl/Trilinos.base/Trilinos/cmake/tribits/ci_support/../../..
  Build Dir: /home/rabartl/Trilinos.base/BUILDS/ATDM/SEMS_RHEL7/CHECKIN/sems-rhel7-cuda-9.2-Pascal60-release-debug
  
  CMake Cache Varibles: -GNinja -DTrilinos_TRIBITS_DIR:PATH=/home/rabartl/Trilinos.base/Trilinos/cmake/tribits -DTrilinos_ENABLE_TESTS:BOOL=ON -DTrilinos_TEST_CATEGORIES:STRING=NIGHTLY -DTrilinos_ALLOW_NO_PACKAGES:BOOL=OFF -DDART_TESTING_TIMEOUT:STRING=600.0 -GNinja -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake -DTrilinos_TRACE_ADD_TEST=ON -DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=ON -DTrilinos_ENABLE_ALL_PACKAGES:BOOL=ON -DTrilinos_ENABLE_ALL_FORWARD_DEP_PACKAGES:BOOL=OFF
  Make Options: -j 20
  CTest Options: -j 10
  
  Pull: Not Performed
  Configure: Passed (1.75 min)
  Build: FAILED (72.73 min)
  Test: FAILED (-1.00 min)
  
  
  ***
  *** WARNING: There are no test results!
  ***
  
  
  Total time for sems-rhel7-cuda-9.2-Pascal60-release-debug = 74.48 min
```

</details>

.

Most of these test failures we have already seen in other builds of Trilinos.  However, the concern is the build failure for the build `sems-rhel7-cuda-9.2-Pascal60-release-debug` which was:

```
[7567/10152] Building CXX object packages/nox/test/epetra/DS6.5.1/CMakeFiles/NOX_DS6_5_1.dir/DennisSchnabel.C.o
...
<built-in>: fatal error: when writing output to : No space left on device
compilation terminated.
```

Hum, looks like we filled the disk up.  I will delete all of the builds and run this last one again.

**Comment by bartlettroscoe** [comment]
FYI: I ran the last build that crashed because of running out of disk space with;

```
$ ./checkin-test-atdm.sh sems-rhel7-cuda-9.2-Pascal60-release-debug \
   --enable-all-packages=on --local-do-all
```

and it returned:

```
FAILED (NOT READY TO PUSH): Trilinos: cortado.sandia.gov

Mon Dec 10 13:28:02 MST 2018

Enabled Packages: 

Build test results:
-------------------
1) sems-rhel7-cuda-9.2-Pascal60-release-debug => FAILED: passed=1998,notpassed=1 => Not ready to push! (209.32 min)
```

<details>
<summary><b>DETAILED BUILD AND TEST RESULTS</b> (click to expand)</summary>

```
1) sems-rhel7-cuda-9.2-Pascal60-release-debug Results:
------------------------------------------------------

  FAILED: Trilinos/sems-rhel7-cuda-9.2-Pascal60-release-debug: passed=1998,notpassed=1
  
  Mon Dec 10 13:28:01 MST 2018
  
  Enabled Packages: 
  Enabled all Packages
  Hostname: cortado.sandia.gov
  Source Dir: /home/rabartl/Trilinos.base/Trilinos/cmake/tribits/ci_support/../../..
  Build Dir: /home/rabartl/Trilinos.base/BUILDS/ATDM/SEMS_RHEL7/CHECKIN/sems-rhel7-cuda-9.2-Pascal60-release-debug
  
  CMake Cache Varibles: -GNinja -DTrilinos_TRIBITS_DIR:PATH=/home/rabartl/Trilinos.base/Trilinos/cmake/tribits -DTrilinos_ENABLE_TESTS:BOOL=ON -DTrilinos_TEST_CATEGORIES:STRING=NIGHTLY -DTrilinos_ALLOW_NO_PACKAGES:BOOL=OFF -DDART_TESTING_TIMEOUT:STRING=600.0 -GNinja -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake -DTrilinos_TRACE_ADD_TEST=ON -DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=ON -DTrilinos_ENABLE_ALL_PACKAGES:BOOL=ON -DTrilinos_ENABLE_ALL_FORWARD_DEP_PACKAGES:BOOL=OFF
  Make Options: -j 20
  CTest Options: -j 10
  
  Pull: Not Performed
  Configure: Passed (1.89 min)
  Build: Passed (135.42 min)
  Test: FAILED (72.01 min)
  
  99% tests passed, 1 tests failed out of 1999
  
  Subproject Time Summary:
  Amesos2          = 115.67 sec*proc (7 tests)
  Anasazi          = 685.44 sec*proc (74 tests)
  Belos            = 681.66 sec*proc (96 tests)
  Ifpack2          = 1021.03 sec*proc (45 tests)
  Intrepid2        = 975.33 sec*proc (267 tests)
  Kokkos           = 510.76 sec*proc (27 tests)
  KokkosKernels    = 442.61 sec*proc (8 tests)
  MueLu            = 8280.47 sec*proc (99 tests)
  NOX              = 388.07 sec*proc (105 tests)
  Panzer           = 14852.38 sec*proc (162 tests)
  Phalanx          =  26.20 sec*proc (27 tests)
  Piro             =  14.86 sec*proc (12 tests)
  Rythmos          =  34.52 sec*proc (83 tests)
  SEACAS           =  14.45 sec*proc (20 tests)
  STK              =   2.32 sec*proc (3 tests)
  Sacado           = 191.02 sec*proc (300 tests)
  Stratimikos      =  21.70 sec*proc (39 tests)
  Teko             = 1386.43 sec*proc (19 tests)
  Tempus           = 124.60 sec*proc (80 tests)
  Teuchos          = 101.71 sec*proc (137 tests)
  Thyra            =  87.11 sec*proc (81 tests)
  Tpetra           = 4031.90 sec*proc (189 tests)
  Xpetra           = 626.86 sec*proc (18 tests)
  Zoltan2          = 1233.97 sec*proc (101 tests)
  
  Total Test time (real) = 4320.47 sec
  
  The following tests FAILED:
  	1976 - PanzerAdaptersSTK_MixedPoissonExample-ConvTest-Hex-Order-3 (Timeout)
  Errors while running CTest
  
  Total time for sems-rhel7-cuda-9.2-Pascal60-release-debug = 209.32 min
```

</details>

.

So the only failure was:

```
  The following tests FAILED:
  	1976 - PanzerAdaptersSTK_MixedPoissonExample-ConvTest-Hex-Order-3 (Timeout)
```

which we have seen in other CUDA builds (see #2446 and #2751).

Now I will update the documentation and the update this PR and it should be ready to review and then merge.

NOTE: @fryeguy52 is going to look into a problem with `bash -l` on 'cortado' that is stopping running the ctest -S Jenkins driver.

**Comment by bartlettroscoe** [comment]
FYI: @bathmatt has some changes he wants to make to this branch so I removed `AT: AUTOMERGE` and added `AT: WIP`.

File: cmake/std/atdm/utils/set_build_options.sh [hunk 1]
```diff
@@ -155,3 +174,11 @@ fi
 # 'cuda' or 'pthread', then that will be selected for the Kokkos backend.
 # Otherwise, if one fo these are not selected and 'openmpi' is present in the
 # build name, then the default Kokkos backend will become 'openmp'!
+
+# Set 'static' or 'shared'
+ATDM_CONFIG_SHARED_LIBS=OFF
```
Comment (jmgate) [inline cmake/std/atdm/utils/set_build_options.sh]: Looks like this line may not be needed given line 20 above.

Comment (bartlettroscoe) [reply cmake/std/atdm/utils/set_build_options.sh]: Thank is true.  Just being more explicit.  I guess the better approach would be to unset those vars at the top of the file and then set the default just before the argument parsing.  Would that be more clear?

Comment (jmgate) [reply cmake/std/atdm/utils/set_build_options.sh]: Whichever you prefer is fine.  Just noticed the duplication and worried about some point in the future where one line gets changed but not  the other, etc.

**Comment by bartlettroscoe** [comment]
FYI: After adding support for 'shared' and 'static' in commits 567c270 and 37ad24c I tested this a little on 'sems-rhel6' with:

```
$ ./checkin-test-atdm.sh \
  gnu-openmp-shared-debug gnu-openmp-static-debug \
 --enable-packages=Kokkos,Teuchos,STK --local-do-all
```

NOTE: This does change behavior for other systems as well as it now uses `*.a` files for the default static builds for some Boost libs on all systems instead of a mix of `*.so` and `*.a` files for static builds like before.  

We will just need to cross out fingers that this does not cause a problem.

**Pull request merged**